### ISSUE TITLE
The variable settings in `config_variables.sh`

### ISSUE THREAD

**Issue by skxooo** [BUTSpeechFIT/EEND_dataprep#12]
1. Could you please clarify which folder the `wav.scp` file path in the RIRS_SCP is coming from: `largeroom`, `mediumroom`, or `smallroom`? Also, does the `wav.scp` file refer to the `rir_list` (as shown below)?（The folder structure of the RIRs dataset is as follows:）

--rir-id large-Room001-00001 --room-id large-Room001 RIRS_NOISES/simulated_rirs/largeroom/Room001/Room001-00001.wav  
--rir-id large-Room001-00002 --room-id large-Room001 RIRS_NOISES/simulated_rirs/largeroom/Room001/Room001-00002.wav  
--rir-id large-Room001-00003 --room-id large-Room001 RIRS_NOISES/simulated_rirs/largeroom/Room001/Room001-00003.wav

RIRS_NOISES
├── pointsource_noises
├── real_rirs_isotropic_noises
└── simulated_rirs
    ├── largeroom
    ├── mediumroom
    └── smallroom

2. Could you please specify which subfolder of the MUSAN noise dataset is being used? What is the format of the scp file?（The folder structure of the MUSAN NOISE dataset is as follows:）

musan
├── music
│   ├── fma
│   ├── fma-western-art
│   ├── hd-classical
│   ├── jamendo
│   └── rfm
├── noise
│   ├── free-sound
│   └── sound-bible
└── speech
    ├── librivox
    └── us-gov

Thank you for your help!

**Comment by skxooo** [comment]
I have one more question. I can't hear any sound from the approximately 2-second long .wav files under my "simulated rirs" folder. Could this be an issue with the dataset? The files are still around tens of KB in size, so they seem to contain data, but there is no audible sound.

**Comment by skxooo** [comment]
> And may I ask how long it took to generate the 2048-hour simulated dataset based on LibriSpeech?

In the solved [issues,](https://github.com/BUTSpeechFIT/EEND_dataprep/issues/5#issuecomment-1784541772) I found that generating the Librispeech dataset takes approximately over 10 hours. Based on this, I have a new question I would like to ask you: In your paper "MULTI-SPEAKER AND WIDE-BAND SIMULATED CONVERSATIONS AS TRAINING DATA FOR END-TO-END NEURAL DIARIZATION," you mentioned using SC training data generated from Librispeech to train a speaker diarization model. Could you please share what your experimental setup was like? How many GPUs did you use, and what models were they? How long did the training process take? This information would be very helpful for my research. I look forward to your reply. Thank you!

**Comment by fnlandini** [comment]
Hi,
These are the scps (change txt by scp, GitHub doesn't like a file if the extension in the name is scp...)
[musan_noise_bg_wav.txt](https://github.com/user-attachments/files/18469653/musan_noise_bg_wav.txt)
[RIRs_wav.txt](https://github.com/user-attachments/files/18469652/RIRs_wav.txt)
It's fine if you don't hear anything on the RIR files. However, if you look at the spectrogram, you should see something. The effect will be applied to the file.
The models in that paper are mentioned, basically EEND-EDA, . I used only 1 GPU for those trainings. I do not remember how long it took, sorry. But it was less than 48 hours even with a small GPU.
I used https://github.com/BUTSpeechFIT/EEND and this was the configuration 

[train.txt](https://github.com/user-attachments/files/18469681/train.txt)

I hope this helps.

**Comment by skxooo** [comment]
Thank you so much for your help! I am now generating simulated conversations, and they sound very much like real-life dialogues. Out of curiosity and to better prepare for future voice activity detection on additional datasets, I was wondering if you could kindly provide guidance on how to use the model available at the link http://kaldi-asr.org/models/m4. Thank you in advance!

**Comment by skxooo** [comment]
May I kindly ask if it is possible for you to provide the specific DER values, both before and after fine-tuning, for SC LibriSpeech (red bars) on the CTS, Maptask, Socio lab, Clinical, AMI H, AMI A, and VoxConv datasets as presented in Figure 1 of the paper 'MULTI-SPEAKER AND WIDE-BAND SIMULATED CONVERSATIONS AS TRAINING DATA FOR END-TO-END NEURAL DIARIZATION'?

**Comment by skxooo** [comment]
I successfully generated the train and validation datasets, but the files generated in the data folder under the test3h directory are empty. The folder v1_2480.6hours/--no-use-rirs--use-noises/durations/ is also empty. The script output is:
cat: '/data/dataset/LibriSpeech_SC/v1_2480.6hours/--no-use-rirs--use-noises/durations/duration_train_*': No such file or directory cat: '/data/dataset/LibriSpeech_SC/v1_2480.6hours/--no-use-rirs--use-noises/durations/duration_validation_*': No such file or directory Defining test set ----- TEST SET DEFINED ----- ----- END -----

**Comment by skxooo** [comment]
Additionally, the file /data/dataset/LibriSpeech_SC/v1_2480.6hours/--no-use-rirs--use-noises/train/data/reco2dur is 0 bytes in size.

**Comment by fnlandini** [comment]
Hi @skxooo sorry for the delay
To run the VAD I used a script like 
```
#!/bin/bash

. ./cmd.sh
. ./path.sh

sad_num_jobs=25
sad_opts="--extra-left-context 79 --extra-right-context 21 --frames-per-chunk 150 --extra-left-context-initial 0 --extra-right-context-final 0 --acwt 0.3"
sad_graph_opts="--min-silence-duration=0.03 --min-speech-duration=0.3 --max-speech-duration=10.0"
sad_priors_opts="--sil-scale=0.1"

data_dir=<your_path>/LibriSpeech_v1/Kaldidatadir

#for set in dev_clean_persession alltrain_persession; do
for set in dev_clean_persession_8k alltrain_persession_8k; do
	# Automatic segmentation using pretrained SAD model
	sad_nnet_dir=exp/segmentation_1a/tdnn_stats_asr_sad_1a
	sad_work_dir=exp/segmentation_1a/tdnn_stats_asr_sad_1a
	if [ ! -d exp/segmentation_1a ]; then
	    wget http://kaldi-asr.org/models/4/0004_tdnn_stats_asr_sad_1a.tar.gz
	    tar zxf 0004_tdnn_stats_asr_sad_1a.tar.gz
	fi
	steps/segmentation/detect_speech_activity.sh \
	    --nj $sad_num_jobs \
	    --graph-opts "$sad_graph_opts" \
	    --transform-probs-opts "$sad_priors_opts" $sad_opts \
	    $data_dir/$set $sad_nnet_dir mfcc_hires $sad_work_dir \
	    $sad_work_dir/$set || exit 1
done
```
You will need to prepare the data directories Kaldi style.


Regarding the results:

librispeech_beforeFT
CTS: 20.77
Maptask: 40.48
Socio lab: 32.47
Clinical: 68.17
AMI H: 35.28
AMI A: 45.1
VoxConverse: 10.52

librispeech_afterFT
CTS: 12.83
Maptask: 12.26
Socio lab: 14.69
Clinical: 22.78
AMI H: 20.31
AMI A: 30.11
VoxConverse: 10.45

**Comment by skxooo** [comment]
Thanks for your help!  I have successfully generated 2,480.6 hours of simulated conversations involving two speakers based on the Librispeech dataset and constructed three datasets: train, validation, and test3h.  

I would like to ask a few questions:  

1. In the SC dataset, is the validation set used for fine-tuning across different speakers?  
2. When should the subset of the validation set be used?  
3. Have you tested the model trained on the 2,480.6-hour Librispeech SC on the test3h dataset? If so, how did it perform?  

Currently, I have not yet processed real-world data, so I can only conduct evaluations on the test3h dataset for now. I would greatly appreciate your valuable insights. 

Additionally, I have a question regarding a training detail: Did you precompute and save the STFT features for the training dataset chunks in advance? I noticed that performing chunking and feature extraction during training significantly slows down the training process.

**Comment by fnlandini** [comment]
That's great!

1,2. The validation set is used during training to check the performance of the model after each epoch on a different set where speakers are different from those in the training set. 
3. I only used the test set at the very beginning but then I simply focused on results on real data. The performance on the test set should be similar to what you see on the validation set because simulated conversations are generated in the same way. Just that the speakers are different.

I tried saving features and it helped a bit in terms of speed but it was not a big difference. Probably more efficient things could be done and I agree that the data loading part of the code is quite slow.

**Comment by skxooo** [comment]
I have understood the role of validation, which is to evaluate the generalization ability of the model during training (i.e., its performance on different speaker datasets). Additionally, in your paper "MULTI-SPEAKER AND WIDE-BAND SIMULATED CONVERSATIONS AS TRAINING DATA FOR END-TO-END NEURAL DIARIZATION," I noticed that Table 2 presents adaptation experiments between different speakers. May I understand that the validation dataset was used during the adaptation process? #

![Image](https://github.com/user-attachments/assets/e484b7f0-97ea-42ce-bcf4-ccfd987832b7)

**Comment by skxooo** [comment]
And I encountered some issues while processing the CallHome dataset. I have processed the CallHome dataset based on parts of the Kaldi recipe, and on this basis, generated the WAV files for CH1 and CH2 using [part1/part1_2spks.txt](https://github.com/BUTSpeechFIT/CALLHOME_sublists/blob/main/part1/part1_2spks.txt) and [part2/part2_2spks.txt](https://github.com/BUTSpeechFIT/CALLHOME_sublists/blob/main/part2/part2_2spks.txt). However, the official spk2utt file does not differentiate between the speakers in each audio file (e.g., in iaaa.sph, where two speakers are present). Currently, I have the complete fullref.rttm file (generated by Kaldi scripts) and the segments file (also generated by Kaldi scripts), but I am unsure how to generate the utt2spk and spk2utt files. I found the script kaldi/egs/chime6/s5b_track2/local/convert_rttm_to_utt2spk_and_segments.py in Kaldi, but it requires a reco2file_and_channel file, and I am not sure how to obtain it. Could you kindly assist me with this issue? Alternatively, could you provide the relevant data files for the dataset?  Thank you in advance for your kind response.

parts of spk2utt file：
iaaa iaaa_000 iaaa_001 iaaa_002 iaaa_003 iaaa_004 iaaa_005 iaaa_006 iaaa_007 iaaa_008 iaaa_009 iaaa_010 iaaa_011 iaaa_012 iaaa_013 iaaa_014 iaaa_015 iaaa_016 iaaa_017 iaaa_018 iaaa_019 iaaa_020 iaaa_021 iaaa_022
iaac iaac_000 iaac_001 iaac_002 iaac_003 iaac_004 iaac_005 iaac_006 iaac_007 iaac_008 iaac_009 iaac_010 iaac_011
iaah iaah_000 iaah_001 iaah_002 iaah_003 iaah_004 iaah_005 iaah_006 iaah_007 iaah_008
iaak iaak_000 iaak_001 iaak_002 iaak_003 iaak_004 iaak_005 iaak_006 iaak_007 iaak_008 iaak_009 iaak_010 iaak_011 iaak_012 iaak_013 iaak_014 iaak_015 iaak_016 iaak_017 iaak_018 iaak_019 iaak_020 iaak_021 iaak_022 iaak_023 iaak_024 iaak_025 iaak_026 iaak_027 iaak_028 iaak_029 iaak_030 iaak_031 iaak_032 iaak_033 iaak_034 iaak_035 iaak_036 iaak_037 iaak_038 iaak_039 iaak_040 iaak_041 iaak_042 iaak_043 iaak_044 iaak_045 iaak_046 iaak_047 iaak_048 iaak_049 iaak_050 iaak_051 iaak_052 iaak_053 iaak_054 iaak_055 iaak_056 iaak_057 iaak_058 iaak_059 iaak_060 iaak_061 iaak_062 iaak_063 iaak_064 iaak_065 iaak_066 iaak_067 iaak_068 iaak_069 iaak_070 iaak_071 iaak_072 iaak_073 iaak_074 iaak_075

**Comment by fnlandini** [comment]
Hi, sorry for the delay
For the adaptation experiments, different training sets were used. You can see in the caption that you shared that those sets are in the order of thousands of hours of audio. In contrast. validation sets are usually fairly small. If you want to create those adaptation (training) sets, you will need to generate them with the corresponding numbers of speakers. For each training set the scripts will also create validation sets that can be used to monitor the training.
Regarding the CallHome sets, please find attached the data directories. I think the should contain what you need. You will need to update the wav.scp so that they point to the paths where you have the data.

[callhome_data.tar.gz](https://github.com/user-attachments/files/18812138/callhome_data.tar.gz)
